v1/nix/modules/drv-parts/nodejs-node-modules/default.nix

```diff
@@ -0,0 +1,39 @@
+{
+  config,
+  lib,
+  dream2nix,
+  ...
+}: let
+  l = lib // builtins;
+
+  cfg = config.nodejs-devshell;
+in {
+  imports = [
+    dream2nix.modules.drv-parts.mkDerivation
+    dream2nix.modules.drv-parts.nodejs-package-lock
+    dream2nix.modules.drv-parts.nodejs-granular
+  ];
+
+  mkDerivation = {
+    dontUnpack = true;
+    dontPatch = true;
+    dontBuild = true;
+    dontInstall = true;
+    dontFixup = true;
+    preBuildPhases = l.mkForce [];
+    preInstallPhases = l.mkForce [];
```

Comment:
I am confused: We import the more complex modules, nodejs-package-lock and nodejs-granular here, just to disable most of their functionality?

Wouldn't that be cleaner the other way round? i.e. basing the imported modules on this one?

Reply:
Generally yes, refactoring the design like that sounds like it would make sense. But currently I'm unsure how much work that implies.

The granular builder does involve more than just defining phases, like, for example computing the datastructure for dependencies correctly. We'd have to factor out that logic first.

If we want to proceed bulding ontop of the granular builder logic, we should probably split up the builder module into several smaller ones. For example we would probably need a `nodejs-dependencies` module to share dependency logic between different other modules.